armi/utils/asciimaps.py

```diff
@@ -40,236 +60,428 @@ class AsciiMap(object):
-        grid = {}
-        # The length of the largest string contents. Useful for spacing everything out
-        # nicely
-        maxSize = len(str(next(iter(self.lattice.values()))))
-        for (i, j), val in self.lattice.items():
-            maxSize = max(len(str(val)), maxSize)
-            x = i * 2
-            y = j * 2 + i
-            grid[x, y] = val
-
-        nRows = max(y for _x, y in grid)
-        minX = min(self._getRowBase(row)[0] for row in range(nRows + 1))
-
-        spacer = " " * maxSize
-
-        for row in reversed(range(nRows + 1)):
-            base = self._getRowBase(row)
-            foundContents = False
-
-            padding = self._getIndices(base, 0)[0] - minX
-            rowContents = []
-            for col in itertools.count(0):
-                i, j = self._getIndices(base, col)
-
-                contents = self.lattice.get((i, j), None)
-                if contents is None and foundContents:
-                    # we have gotten to the end of the contents for this row; break out
-                    # of our otherwise infinite loop
-                    break
-                if contents is None:
-                    contents = PLACEHOLDER
-                elif not foundContents:
-                    foundContents = True
-
-                rowContents.append(contents.center(maxSize))
-
-            stream.write(" " * padding * maxSize + spacer.join(rowContents) + "\n")
+        return i, j
+
+    def _makeOffsets(self):
+        """
+        Handle offsets for full-hex flat grids.
+        
+        Due to the staggered nature, these have 0 or 1 offsets on 
+        top and and then 0 or 1 + an actual offset on the bottom.
+        """
+        # max lines required if corners were not cut off
+        maxIJIndex = self._ijMax
+        self.asciiOffsets = []
+        # grap top left edge going down until corner where it lifts off edge.
+        # Due to the placeholders these just oscillate
+        for li in range(maxIJIndex * 3):
+            self.asciiOffsets.append((li - self._asciiLinesOffCorner) % 2)
+
+        # going away from the left edge, the offsets increase linearly
+        self.asciiOffsets.extend(range(maxIJIndex + 1))
+
+        # since we allow cut-off corners, we must truncate the offsets
+        # number of items in last line indicates how many
+        # need to be cut. (first line has placeholders...)
+        cutoff = self._asciiLinesOffCorner
+        if cutoff:
+            self.asciiOffsets = self.asciiOffsets[cutoff:-cutoff]
+
+    def _updateDimensionsFromData(self):
+        AsciiMapHexThirdFlatsUp._updateDimensionsFromData(self)
+        self._asciiMaxCol = self._ijMax + 1
+        self._asciiMaxLine = self._ijMax * 4 + 1 - self._asciiLinesOffCorner * 2
 
 
 class AsciiMapHexFullTipsUp(AsciiMap):
     """
-    Read a full hexagonal ASCII map with tips of hexagons pointing up.
+    Full hex with tips up of the smaller cells. 
```

Comment:
...huh? "Full-hex geometry, with hex corners oriented vertically" maybe. Not sure what "of the smaller cells" means